### PR TITLE
feat: Add editor config & eslint configs

### PR DESCRIPTION
**Pull request by njfamirm**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,13 +5,29 @@
   "repository": "https://github.com/AliMD/One",
   "author": "Ali Mihandoost <alimihandoost@gmail.com>",
```
Comment (mm25zamanian) [inline package.json:6]: please change author from `Ali Mihandoost <alimihandoost@gmail.com>` to `@one community`

**Review by mm25zamanian — CHANGES_REQUESTED** [review]
first: please check my pull request !
and second: change settings and config with my requests and comments

**Comment by njfamirm** [comment]
I coordinated with @asmanp2012

File: package.json (hunk 1 above)
Comment (njfamirm) [reply package.json:6]: I do not dare 🤣

File: .gitignore [hunk 2]
```diff
@@ -15,9 +15,11 @@ Desktop.ini
...
 dist
-node_modules
 .tsbuildinfo
 tsconfig.tsbuildinfo
+
+# Dependencies
+node_modules
 package-lock.json
 
 *.js
```
[the remark this answers is not in the document]
Comment (njfamirm) [reply .gitignore:25]: use commmment

File: package.json (hunk 1 above)
Comment (mm25zamanian) [reply package.json:6]: 😂😂🤣
but @asmanp2012 say do it (in meet) !

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,26 @@
+// Reference: eslint.org/docs/user-guide/configuring/configuration-files
+
+{
+  "compilerOptions": {
+    "incremental": true,
+    "target": "ESNEXT",
+    "module": "ESNEXT",
+    "lib": ["ESNEXT", "DOM", "DOM.Iterable"],
```
[the remark this answers is not in the document]
Comment (asmanp2012) [reply .eslintrc:8]: @njfamirm remove `.eslintrc` file

**Comment by asmanp2012** [comment]
> first: please check my pull request ! and second: change settings and config with my requests and comments

You should merge and fix the conflict.

**Review by asmanp2012 — APPROVED** [review]
Thanks

**Pull request merged**